### PR TITLE
feat(core,schemas): add daily token usage table to ease the usage calculation

### PR DESCRIPTION
**Pull request by darcyYe**
## Summary

add daily token usage table to ease the usage calculation.
Token usage includes:
1. issue of access token and opaque access token (for both users and applications)
2. issue of refresh token

## Testing

Tested locally.
Will add integration tests to cloud since the token usage is concerned there.

## Checklist

- [ ] `.changeset`
- [x] unit tests
- [ ] integration tests
- [x] necessary TSDoc comments

### CODE REVIEW

File: packages/core/src/event-listeners/token-usage-counter.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import type Queries from '#src/tenants/Queries.js';
+
+export const tokenUsageCounter = async (queries: Queries) => {
+  const { recordTokenUsage } = queries.dailyTokenUsage;
+
+  // Mark this user as active today
```
Comment (wangsijie) [inline packages/core/src/event-listeners/token-usage-counter.ts:6]: nit: change the comment

File: packages/core/src/event-listeners/index.ts [hunk 2]
```diff
@@ -21,6 +22,9 @@ export const addOidcEventListeners = (provider: Provider, queries: Queries) => {
   provider.addListener('access_token.saved', async (token) => {
     return accessTokenIssuedListener(token, queries);
   });
+  provider.addListener('client_credentials.saved', async () => tokenUsageCounter(queries));
+  provider.addListener('client_credentials.issued', async () => tokenUsageCounter(queries));
+  provider.addListener('refresh_token.saved', async () => tokenUsageCounter(queries));
```
Comment (wangsijie) [inline packages/core/src/event-listeners/index.ts:27]: nit: this may cause duplicate counts, maybe we can align one type of token to record

**Review by wangsijie — APPROVED** [review]

File: packages/core/src/queries/daily-token-usage.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    // should be the first token use of the day), otherwise increment the usage by 1.
+    pool.one<DailyTokenUsage>(sql`
+      insert into ${table} (${fields.id}, ${fields.date}, ${fields.usage})
+      values (${generateStandardId()}, to_timestamp(${getUtcStartOfTheDay(
+        date
+      ).getTime()}::double precision / 1000), 1)
+      on conflict (${fields.date}, ${fields.tenantId}) do update set ${fields.usage} = ${
+        fieldsWithPrefix.usage
+      } + 1
+      returning *
```
Comment (gao-sun) [inline packages/core/src/queries/daily-token-usage.ts:40]: better use `${Obejct.values(fields)}`

Comment (darcyYe) [reply packages/core/src/queries/daily-token-usage.ts]: Updated

File: packages/core/src/event-listeners/index.ts (hunk 2 above)
Comment (darcyYe) [reply packages/core/src/event-listeners/index.ts]: Per offline discussion, count refresh token for now.

**Review by gao-sun — APPROVED** [review]

**Pull request merged**